functions/pupil-status/index.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const azureStorage = require('azure-storage')
+const bluebird = require('bluebird')
+const moment = require('moment')
+const sqlService = require('less-tedious')
+const uuid = require('uuid/v4')
+const winston = require('winston')
+const { TYPES } = require('tedious')
+
+winston.level = 'error'
+const config = require('../config')
+sqlService.initialise(config)
+
+const schema = '[mtc_admin]'
+let azureTableService
+initAzureTableService()
+
+module.exports = async function (context, pupilStatusMessage) {
+  context.log('pupil-status message received')
+
+  // TODO process pupil status
+
+  // Store the raw message to an audit log
+  context.bindings.pupilEventsTable = []
+  const entity = {
+      // TODO will we always have a check code?
+    PartitionKey: pupilStatusMessage.checkCode,
+    RowKey: uuid(),
+    eventType: 'pupil-status',
+    payload: JSON.stringify(pupilStatusMessage),
+    processedAt: moment().toDate()
+  }
+
+  context.bindings.pupilEventsTable.push(entity)
+}
+
+/**
+ * Promisify the azureStorage library as it still lacks Promise support
+ */
+function initAzureTableService () {
+  if (!azureTableService) {
+    azureTableService = azureStorage.createTableService()
+    bluebird.promisifyAll(azureTableService, {
+      promisifier: (originalFunction) => function (...args) {

```

Comment:
Similar comment I made to Jon's PR, why not promisifying what is required for the implementation?
The time needed for the whole 'promisification' might be small but the library consists of many methods that we are promisifying for no reason as they will not be used.

Reply:
this happens once when the functions app (containing ALL of the functions) is loaded.  
it makes sense to have one piece of code that promisifies the entire library, rather than boilerplate for each method.
However - could we extract this out to a file in the functions root folder, and then it is loaded once for ALL functions instead of in each function.  This would be similar to how the sql connection pooling works i believe.  @jon-shipley thoughts?

Reply:
Do you mean this?
https://github.com/DFEAGILEDEVOPS/MTC/pull/819/files#diff-e43640ff0b6d9faccb1709042885f09c

Reply:
thats what i was thinking, perfect 👍  
@GeorgeChatzigiannis i will implement this now

Reply:
👍